aligner/u_of_michigan_aligner/u_of_michigan_aligner.wdl

```diff
@@ -11,152 +12,167 @@
-     additional_disk else dbSNP_disk_size_override_default + additional_disk
-
-  Float cram_and_crai_size = if (defined(dynamically_calculate_disk_requirement)) then size(input_cram_file, "GB") + size(input_crai_file, "GB") + 
-     additional_disk else CRAMandCRAI_disk_size_override_default + additional_disk
-
-  Float fastq_gz_files_size = CRAM_to_fastqgz_multiplier * cram_and_crai_size
-
-
-
-  Float PreAlign_disk_size = PreAlign_ref_size + (bwa_disk_multiplier * cram_and_crai_size) + 
-     (sort_sam_disk_multiplier * cram_and_crai_size) + cram_and_crai_size + additional_disk + fastq_gz_files_size
-
-  Float Align_disk_size = ref_size + ref_extra_size + (bwa_disk_multiplier * fastq_gz_files_size) + additional_disk
-
-  # The merged cram can be bigger than the summed sizes of the individual aligned crams,
-  # so account for the output size by multiplying the input size by bwa disk multiplier.
-  Float PostAlign_disk_size = ref_size + dbsnp_size + cram_and_crai_size + 
-     (sort_sam_disk_multiplier * cram_and_crai_size) + (bwa_disk_multiplier * cram_and_crai_size) + additional_disk
-
+  input {
+
+    # The first nine variables are neccessary to convert a float to int
+    # https://gatkforums.broadinstitute.org/wdl/discussion/9541/convert-float-to-int
+    # The old version, disks: "local-disk " + sub(disk_size, "\\..*", "") + " HDD"
+    # being set as as in input for the task's inputs, previously worked but fails to 
+    # pass womtool in WDL 1.0 format; documentation on this change is absent
+
+    String PreAlign_disk_size_string = PreAlign_disk_size
+    String PreAlign_disk_size_before_decimal = sub(PreAlign_disk_size_string, "\\..*", "")
+    Int PreAlign_disk_size_int = PreAlign_disk_size_before_decimal
+
+    String Align_disk_size_string = Align_disk_size
+    String Align_disk_size_before_decimal = sub(Align_disk_size_string, "\\..*", "")
+    Int Align_disk_size_int = Align_disk_size_before_decimal
+
+    String PostAlign_disk_size_string = PostAlign_disk_size
+    String PostAlign_disk_size_before_decimal = sub(PostAlign_disk_size_string, "\\..*", "")
+    Int PostAlign_disk_size_int = PostAlign_disk_size_before_decimal
+
+    File? input_crai_file
+    File input_cram_file
+
+    String docker_image
+
+    File ref_alt
+    File ref_bwt
+    File ref_pac
+    File ref_ann
+    File ref_amb
+    File ref_sa
+
+    File ref_fasta
+    File ref_fasta_index
+
+    File dbSNP_vcf
+    File dbSNP_vcf_index
+
+    # The CRAM to be realigned may have been aligned with a different reference
+    # genome than what will be used in the alignment step. The pre align step
+    # must use the reference genome that the CRAM was originally aligned with
+    # to convert the CRAM to a SAM
+    File? PreAlign_reference_genome
+    File PreAlign_reference_genome_default = select_first([PreAlign_reference_genome,ref_fasta])
```

Comment:
I think with WDL 1.0 you can simplify a default setting for a variable like this:
```
File PreAlign_reference_genome_default = ref_fasta
```
I think in this case if PreAlign_reference_genome_default is not provided in the input json it uses ref_fasta; but you will need to verify this. See https://github.com/DataBiosphere/topmed-workflow-variant-calling/blob/master/variant-caller-wdl/topmed_freeze8_caller.wdl

https://github.com/openwdl/wdl/blob/master/versions/1.0/SPEC.md#declared-inputs-defaults-and-overrides

Reply:
In my opinion it's better to leave it as-is because the logic of the new option is harder to follow -- ie, it's not immediately obvious that it is a default that can be overwritten, while with `select_first()` both situations are immediately clear.